web/channels/player_channel.ex

```diff
@@ -34,7 +34,12 @@ defmodule Yggdrasil.PlayerChannel do
   end
 
   def handle_in("player_cmd", %{ "text" => text }, socket) do
-    Player.run_cmd(socket.assigns.user, Message.command(text))
-    {:noreply, socket}
+    case Parser.parse(text) do
+      {:ok, cmd} ->
+        Player.run_cmd(socket.assigns.user, cmd)
+        {:noreply, socket}
+      {:error, reason} ->
+        {:reply, {:error, Message.error(reason)},  socket}
```

Comment:
Yeah that's a great idea. I didn't think about what that status might translate to on the client. I was just trying to get everything working together on the server.